api/src/main/java/api/UserAnswerResource.java

```diff
@@ -52,4 +53,28 @@
     @DELETE
     @Path("me/answers/{answerId}")
     Observable<Void> deleteAnswer(Auth auth, @PathParam("answerId") long answerId);
+
+    /**
+     * Upvotes (+1) an answer.
+     * Upvoting an existing downvote will result in a neutral (0) vote.
+     * Upvoting an existing upvote will result in error.
+     * @param auth
+     * @param answerId
+     * @return
+     */
+    @POST
+    @Path("me/answers/{answerId}/upvote")
```

Comment:
Consider PUT/me/answers/{answerId}/vote sending the representation of what the vote should be. There will then be a representation of that vote available on GET /me/answers/{answerId}/vote.

Reply:
That's a good idea, and more "restful". Future improvement?